src/sagemaker_containers/_trainer.py

```diff
@@ -59,7 +68,12 @@ def train():
             entry_point.run(env.module_dir, env.user_entry_point, env.to_cmd_args(), env.to_env_vars())
 
         logger.info('Reporting training SUCCESS')
+
         files.write_success_file()
+
+        if intemediate_sync:
```

Comment:
in multiple branches. can we consolidate in one finally?

Reply:
i don't think so :(
because we need to exist with different exit codes the main process, and we can't do anything after we finished main process...

Reply:
ok